src/core/georeferencing.h

```diff
@@ -717,6 +735,20 @@ double Georeferencing::roundDeclination(double value)
 	return floor(value*100.0+0.5)/100.0;
 }
 
+inline
+constexpr unsigned int Georeferencing::grivationPrecision()
+{
+	// This must match the implementation in grivationRound().
+	return 8u;
+}
+
+inline
+double Georeferencing::roundGrivation(double value)
+{
+	// This must match the implementation in grivationPrecision().
+	return floor(value*1E8+0.5)/1E8;
+}
```

Comment:
(optional code enhancement)
I like that the grivation rounding implementation is coherent with the declination rounding. What stands out in both places is the open possibility for inconsistency reflected in the two comments.

My proposal would be:
```suggestion
inline
constexpr unsigned int Georeferencing::grivationPrecision()
{
	return 8u;
}

inline
double Georeferencing::roundGrivation(double value)
{
	constexpr auto multiplier = pow(10, grivationPrecision());
	return floor(value*multiplier+0.5)/multiplier;
}
```
The same pattern would apply to declination rounding. If that weren't a header file, a common rounding function would be beneficial. At this place, the new symbol would pollute the namespace, though.

Reply:
I like your method of eliminating the need for the comment, though I don't trust `pow` to return an integral value. 

`constexpr auto multiplier = floor(pow(10, grivationPrecision()) +0.5)`

would work.

Reply:
Please note that `pow` isn't `constexpr` by the standard. That's why I used the comment in `roundDeclination`.
https://en.cppreference.com/w/cpp/numeric/math/pow 

A compile-time solution for `int` is possible with templates (e.g. in an anonymous namespace early in the cpp file):
~~~
template<int N> constexpr int power(int base)
{
	return base * power<N - 1>(base);
}

template<> constexpr int power<0>(int)
{
	return 1;
}

static_assert(power<0>(10) == 1, "10^0");
static_assert(power<3>(10) == 1000, "10^3");
~~~

Reply:
I was leaning toward keeping the PR as is, thinking that the template adds complexity that is not justified by the small risk of the precision and rounding methods getting out of alignment.

However, the `power()` template makes it possible to eliminate the literal `1E6` that appears in the definition of 'getDeclinationSetting()`. That literal is another value which needs to be kept coordinated with the precisions, and which I had not taken the trouble to comment because it would need to be mentioned in 3 places. So I will add the template suggested by @dg0yt in order to statically verify the relationships that needed comments. I'll push a revision to the branch with that change.

It was necessary to introduce protected methods in the `Georeferencing` class to return the "multiplier" contstants that are used by the `roundXxxxxx()` methods, in order to express `static_assert`s for the relationships.

Reply:
Implemented the template in f530fcf, applying it to `scaleFactorPrecision()` and `declinationPrecision()`. The comments indicating how the implementations must match are replaced with a `static_assert` in `georeferencing.cpp`.

Reply:
Good point about the non-standard `constexpr` behavior of `pow`., Kai.

I like the static checks in `georeferencing.cpp` as it spots the potential discrepancy at compile time, Scott. I cannot tell, however, how Kai will approach the change.